pkg/controllers/management/auth/user_attribute_handler.go

```diff
@@ -76,6 +78,14 @@ func (c *UserAttributeController) sync(key string, attribs *v3.UserAttribute) (r
 
 	attribs, err = c.providerRefresh(attribs)
 	if err != nil {
+		var retrieveErr *oauth2.RetrieveError
+		// Stop retrying if the token has expired.
+		if errors.As(err, &retrieveErr) {
+			if retrieveErr.ErrorCode == "invalid_grant" && retrieveErr.ErrorDescription == "Token is not active" {
```

Comment:
RFC6749 implies that the token is "invalid" which I guess means that it _probably_ won't become "valid" any time soon.

So, yeah, I think checking for `invalid_grant` is good, it would probably be useful to debug log out _why_ in case the server returns more (the description is meant to be human-readable).